x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunker.java

```diff
@@ -39,15 +40,23 @@
 public class EmbeddingRequestChunker<E extends EmbeddingResults.Embedding<E>> {
 
     // Visible for testing
-    record Request(int inputIndex, int chunkIndex, ChunkOffset chunk, List<String> inputs) {
+    record Request(int inputIndex, int chunkIndex, ChunkOffset chunk, List<ChunkInferenceInput> inputs) {
         public String chunkText() {
-            return inputs.get(inputIndex).substring(chunk.start(), chunk.end());
+            return inputs.get(inputIndex).input().substring(chunk.start(), chunk.end());
+        }
+
+        public ChunkInferenceInput chunkInput() {
+            ChunkInferenceInput chunkInferenceInput = inputs.get(inputIndex);
+            String chunkText = chunkInferenceInput.input().substring(chunk.start(), chunk.end());
+            return new ChunkInferenceInput(chunkText, chunkInferenceInput.chunkingSettings());
         }
     }
 
     public record BatchRequest(List<Request> requests) {
-        public List<String> inputs() {
-            return requests.stream().map(Request::chunkText).collect(Collectors.toList());
+        public List<ChunkInferenceInput> inputs() {
+            return requests.stream()
+                .map(r -> new ChunkInferenceInput(r.chunkText(), r.inputs().getFirst().chunkingSettings()))
```

Comment:
I'm not sure if this is exactly what we want, but it does help to get tests to pass. I initially did a flat map but that will exponentially increase the size of the returned list.

Reply:
Oh I like that, it's much cleaner. Thank you!